rskj-core/src/main/java/co/rsk/net/light/LightSyncProcessor.java

```diff
@@ -0,0 +1,190 @@
+        this.version = (byte) 0;
+        this.pendingMessages = new LinkedHashMap<Long, LightClientMessageCodes>() {
+            @Override
+            protected boolean removeEldestEntry(Map.Entry<Long, LightClientMessageCodes> eldest) {
+                boolean shouldDiscard = size() > MAX_PENDING_MESSAGES;
+                if (shouldDiscard) {
+                    loggerNet.trace("Pending {}@{} DISCARDED", eldest.getValue(), eldest.getKey());
+                }
+                return shouldDiscard;
+            }
+        };
+    }
+
+    public void processStatusMessage(StatusMessage msg, LightPeer lightPeer, ChannelHandlerContext ctx, LightClientHandler lightClientHandler) {
+        LightStatus status = msg.getStatus();
+        try {
+            String bestHashLog = HashUtil.shortHash(status.getBestHash());
+            loggerNet.debug("Receiving Status - block {} {}", status.getBestNumber(), bestHashLog);
+
+            byte protocolVersion = status.getProtocolVersion();
+            if (protocolVersion != version) {
+                loggerNet.info("Removing LCHandler for {} due to protocol incompatibility", ctx.channel().remoteAddress());
+                loggerNet.info("Protocol version {} - message protocol version {}",
+                        version,
+                        protocolVersion);
+                lightPeer.disconnect(ReasonCode.INCOMPATIBLE_PROTOCOL);
+                ctx.pipeline().remove(lightClientHandler); // Peer is not compatible for the 'lc' sub-protocol
+                return;
+            }
+
+            int networkId = config.networkId();
+            int msgNetworkId = status.getNetworkId();
+            if (msgNetworkId != networkId) {
+                loggerNet.info("Removing LCHandler for {} due to invalid network", ctx.channel().remoteAddress());
+                loggerNet.info("Different network received: config network ID {} - message network ID {}",
+                        networkId, msgNetworkId);
+                lightPeer.disconnect(ReasonCode.NULL_IDENTITY);
+                ctx.pipeline().remove(lightClientHandler);
+                return;
+            }
+
+            Keccak256 genesisHash = genesis.getHash();
+            Keccak256 msgGenesisHash = new Keccak256(status.getGenesisHash());
+            if (!msgGenesisHash.equals(genesisHash)) {
+                loggerNet.info("Removing LCHandler for {} due to unexpected genesis", ctx.channel().remoteAddress());
+                loggerNet.info("Config genesis hash {} - message genesis hash {}",
+                        genesisHash, msgGenesisHash);
+                lightPeer.disconnect(ReasonCode.UNEXPECTED_GENESIS);
+                ctx.pipeline().remove(lightClientHandler);
+                return;
+            }
+        } catch (NoSuchElementException e) {
+            loggerNet.debug("LCHandler already removed - exception: {}", e.getMessage());
+        }
+
+        if (peerStatuses.size() >= MAX_PEER_CONNECTIONS) {
```

Comment:
I don't understand this line nor the usage of the Map. Why use a Map for what essentially could be an int? why you don't disconnect in this case? Shouldn't there be a mechanism to account for disconnections?

Reply:
This is another case of "dummy things". The map is the basic approach to handle peers. For now, there is no something like "PeerHandlers". In the sync stage, peerStatuses is used to get the best peer, I mean, the peer with high total difficult and with the best block.
Then, each node accepts a max quantity of peers.